ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -434,6 +562,233 @@ const triggerSet: TriggerSet<Data> = {
+    {
+      id: 'Hunt Ttokrrone Desert Tempest',
+      type: 'StartsUsing',
+      netRegex: { id: ttokrroneDesertTempestIds, source: 'Ttokrrone', capture: true },
+      durationSeconds: 7,
+      alertText: (data, matches, output) => {
+        const tempest = ttokrroneDesertTempest[matches.id]!;
+        const pattern = data.ttokSandOrbPatterns[data.ttokSandOrbOnSet];
+        let inOut: TtokrroneDodge = 'unknown';
+        let dir2: TtokrroneDodge | null = null;
+        let dir3: TtokrroneDodge | null = null;
+        let backFrontSpot: TtokSafeSpots = 0b111_111_111_111;
+
+        if (tempest & TtokSafeSpots.In) {
+          inOut = 'in';
+        } else if (tempest & TtokSafeSpots.Out) {
+          inOut = 'out';
+        }
+        if (tempest === TtokSafeSpots.InRight) {
+          dir2 = 'right';
+        } else if (tempest === TtokSafeSpots.InLeft) {
+          dir2 = 'left';
+        }
+
+        if (pattern && data.ttokRotated === 0) {
+          const safeSpot = tempest & pattern;
+
+          if (safeSpot & TtokSafeSpots.Back) {
+            backFrontSpot = TtokSafeSpots.Back;
+            dir3 = 'back';
+          } else if (safeSpot & TtokSafeSpots.Front) {
+            backFrontSpot = TtokSafeSpots.Front;
+            dir3 = 'front';
+          }
+          // if the tempest was only in or out, but the sand pattern needs right or left
+          if (!dir2 && !(safeSpot & 0b010_010_010_010)) {
+            // we also need to check the other selected direction (front/back)
+            if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
+              dir2 = 'right';
+            } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
+              dir2 = 'left';
+            }
+          }
+          // swap in case dir2 is null, and its more natural phrasing
+          [dir2, dir3] = [dir3, dir2];
+          data.ttokSandOrbOnSet++;
+        }
+        if (dir2 && dir3) {
+          return output.triple!({
+            inOut: output[inOut]!(),
+            dir2: output[dir2]!(), // front or back
+            dir3: output[dir3]!(), // right or left
+          });
+        }
+        if (dir2) {
+          return output.double!({ inOut: output[inOut]!(), dir2: output[dir2]!() });
+        }
+        return output[inOut]!();
+      },
+      outputStrings: ttokrroneTempestSandspoutOutputStrings,
+    },
+    // Aoe
+    {
+      id: 'Hunt Ttokrrone Touchdown',
+      type: 'StartsUsing',
+      netRegex: { id: '91DB', source: 'Ttokrrone', capture: false },
+      durationSeconds: 4,
+      response: Responses.aoe(),
+      run: (data) => {
+        // Reset everything as orb sequences start with a touchdown.
+        data.ttokSandOrbSets = 0;
+        data.ttokSandOrbOnSet = 0;
+        data.ttokSandOrbs = [];
+        data.ttokSandOrbPatterns = [];
+        data.ttokRotated = 0;
+      },
+    },
+    // Front cleave, rotates C or CCW and cleaves 4 or 7 times (maybe health dependent?)
```

Comment:
I have limited logs so far, but -- unfortunately -- it doesn't seem that the cast id is dispositive of how many cleaves. 

As best as I can tell so far, C9/CA can be either 4 or 7, although if 4, there does usually (always?) seem to be a follow-up set of 4 in the other direction that use C5/C6).  Given that each turn/cleave has a separate 0x15 line, it's likely something that's determined by the fight script and not something we can detect in advance client-side).

Reply:
Yes that does seem to be the case, I was not able to find any differences.